src/node_funcs.c

```diff
@@ -0,0 +1,40 @@
+#include "postgres.h"
+
+#include "pg_diffix/node_funcs.h"
+#include "pg_diffix/utils.h"
+
+static ParamExternData *get_param_data(ParamListInfo bound_params, int one_based_paramid)
+{
+  if (bound_params->paramFetch != NULL)
+    return bound_params->paramFetch(bound_params, one_based_paramid - 1, true, NULL);
+  else
+    return &bound_params->params[one_based_paramid - 1];
+}
+
+bool is_simple_constant(Node *node)
```

Comment:
So we have decided on the term "simple constant" to mean const and param? Maybe use `stable expression` like Postgres? I like that one better.

Reply:
We haven't really decided that, I just like the term for some reason. I avoided any PostgreSQL-loaded terminology, since what matters here is Diffix' POV, but `stable_expression` sounds good too.